validator/sawtooth_validator/journal/publisher.py

```diff
@@ -714,3 +726,25 @@ def has_batch(self, batch_id):
                 return True
             if batch_id in self._queued_batch_ids:
                 return True
+
+
+class _RollingAverage(object):
+
+    def __init__(self, sample_size, initial_value):
+        self._samples = deque(maxlen=sample_size)
+
+        self._samples.append(initial_value)
+        self._current_average = initial_value
+
+    @property
+    def value(self):
+        return self._current_average
+
+    def update(self, sample):
+        """Add the sample and return the updated average.
+        """
+        self._samples.append(sample)
+
+        self._current_average = sum(list(self._samples)) / len(self._samples)
```

Comment:
You do not need to convert deque to a list: https://docs.python.org/3/library/collections.html#deque-recipes